src/compiler/crystal/codegen/ast.cr

```diff
@@ -23,7 +23,7 @@ module Crystal
 
   class NumberLiteral
     def zero?
-      case :kind
+      case kind
```

Comment:
That was a silent bug it seems.

Reply:
If this always matched the `else` branch and didn't break anything, we're either missing a spec or this entire thing is unnecessary.

Reply:
It's unused

Reply:
@straight-shoota We can either remove it altogether or expose it to the macro land (which I'd prefer) via #10248.

Reply:
just leave it like for this PR